### PR TITLE
Ubbabeck security page refactor

### PR DESCRIPTION
**Pull request by ubbabeck**
Tested with remote storage.

@erdemyerebasmaz Do you think this cause unwanted side effects with iCloud or GDrive ?

### CODE REVIEW

File: lib/routes/initial_walkthrough/initial_walkthrough.dart [hunk 1]
```diff
@@ -254,7 +254,6 @@ class _InitialWalkthroughPageState extends State<InitialWalkthroughPage>
 
   void _restoreFromBackup() async {
     log.info("Restore from Backup");
-    await _signOut();
```
Comment (erdemyerebasmaz) [inline lib/routes/initial_walkthrough/initial_walkthrough.dart:257]: Why was this removed?

Comment (ubbabeck) [reply lib/routes/initial_walkthrough/initial_walkthrough.dart:257]: When pressing restore from backup button nothing happens on fresh install. 

Why do we need it in the first place on a fresh install?

File: lib/widgets/enable_backup_dialog.dart [hunk 2]
```diff
@@ -283,7 +279,9 @@ class _BackupNowButtonState extends State<_BackupNowButton> {
               backupProvider: BackupProvider.remoteServer(),
               remoteServerAuthData: auth,
             ),
-          ).then((_) => Navigator.pop(context));
```
Comment (erdemyerebasmaz) [inline lib/widgets/enable_backup_dialog.dart:286]: popUntil wasn't used because backup prompt dialog can be shown on top of any route and we want to pop the dialog itself, not redirect users to home page.

File: lib/widgets/enable_backup_dialog.dart [hunk 3]
```diff
@@ -258,13 +258,9 @@ class _BackupNowButtonState extends State<_BackupNowButton> {
...
   ) async {
     try {
-      if (selectedProvider.isRemoteServer) {
-        await _enterRemoteServerCredentials();
-      } else {
-        await _backupNow(
-          widget.backupSettings.copyWith(backupProvider: selectedProvider),
-        ).then((_) => Navigator.pop(context));
-      }
+      await _backupNow(
```
Comment (erdemyerebasmaz) [inline lib/widgets/enable_backup_dialog.dart:261]: Don't we want users to be able to switch their remote server, re-enter their auth credentials here?

File: lib/routes/initial_walkthrough/initial_walkthrough.dart (hunk 1 above)
Comment (erdemyerebasmaz) [reply lib/routes/initial_walkthrough/initial_walkthrough.dart:257]: We want users to be able to switch their Google account without requiring a re-install.

Let's assume I have 2 google accounts: Account A, B
- I selected Google Drive & then Account A, (by mistake)
- Account A's snapshots are loaded or failed to load
- I cancel the flow, press restore again and select Google Drive
- Account A is now selected by default as it's the signed in account.
- User can't switch their Google account on the installed app anymore.

File: lib/widgets/enable_backup_dialog.dart (hunk 3 above)
Comment (erdemyerebasmaz) [reply lib/widgets/enable_backup_dialog.dart:261]: I now see why this is required, we do prompt for auth data on Select Backup Provider dialog.
